### PR TITLE
test: 68% coverage of my advances page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 7c20f4c</samp>

This pull request adds and updates several files related to the my advances feature. It improves the test coverage, code quality, and mock data for the my advances page component, the extended advance request feature, and the my advances filters feature. It also introduces a new interface for the filters model, `MyAdvancesFilters`, and refactors the existing code to use it.

### <samp>🤖 Generated by Copilot at 7c20f4c</samp>

> _We mock the data for the extended advance_
> _We filter and sort the advances we have_
> _We define the interface for our model_
> _We refactor and fix the code of our hell_

### <samp>🤖 Generated by Copilot at 7c20f4c</samp>

*  Add and modify mock data objects for testing extended advance request and my advances filters features ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-5523a698b796df7860831812b5e4121666f847e9e39be66fa11c5d5dbfe1e577R705-R722), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-a850c84de490df80c1a7fc8d4713fb9a7b420e9539224d90a89aa03833dc618aR73-R128), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-9eba24c00e66be0e747c8ba0d07bbe59789c3f70b2ec1606bd082f8b4ade1fd0R1-R20))
*  Define `MyAdvancesFilters` interface for the filters model of the my advances page ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-ff90548ae456520da1e19cef3e97bc24d92b47f6c1072904c74c2a99baa860f7R1-R9))
*  Remove unused imports and fix typos in `my-advances.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL19-L20), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL41-R34), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL133-R126), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL180-R182), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL249-R243), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL259-R252))
*  Change the type alias and casting of filters from `Filters` to `Partial<MyAdvancesFilters>` in `my-advances.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL28-R27), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL63-R56), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-f2951c001ee49b79e597c3323e7a4ae3e3d3b94861f1f1d7dfc3be1b96fd8e0bL127-R120))
*  Add `filters` and `params` properties to the mock activated route object in `my-advances.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-b5c01b9eeeec345d0700b5cb4a8b0454a8c2244a15278d654fe57982233c0fecL68-R93))
*  Add `ionViewWillEnter()` test suite to test the logic and functionality of the method in `my-advances.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-b5c01b9eeeec345d0700b5cb4a8b0454a8c2244a15278d654fe57982233c0fecR190-R372))
*  Add imports for testing asynchronous operations and timing in `my-advances.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-b5c01b9eeeec345d0700b5cb4a8b0454a8c2244a15278d654fe57982233c0fecL1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2409/files?diff=unified&w=0#diff-b5c01b9eeeec345d0700b5cb4a8b0454a8c2244a15278d654fe57982233c0fecR19-R39))

## Clickup
https://app.clickup.com/t/85ztyue80

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/my-advances/my-advances.page.spec.ts [hunk 1]
```diff
@@ -65,7 +86,10 @@ describe('MyAdvancesPage', () => {
           useValue: {
             snapshot: {
               queryParams: {
-                filter: 'all',
+                filters: JSON.stringify(myAdvancesfiltersData),
```
Comment (OmkarJ13) [inline src/app/fyle/my-advances/my-advances.page.spec.ts:89]: myAdvancesfiltersData -> myAdvancesFiltersData

File: src/app/fyle/my-advances/my-advances.page.spec.ts [hunk 2]
```diff
@@ -163,4 +187,187 @@ describe('MyAdvancesPage', () => {
...
+      spyOn(component.filterParams$, 'next');
+      component.ionViewWillEnter();
+      expect(component.filterParams$.next).toHaveBeenCalledOnceWith(myAdvancesfiltersData);
+      expect(component.filterPills).toEqual(allFilterPills);
+      expect(filtersHelperService.generateFilterPills).toHaveBeenCalledOnceWith(component.filterParams$.value);
+    });
+
+    it('should set myAdvancerequests$ to singleExtendedAdvReqRes.data', () => {
+      component.ionViewWillEnter();
+      component.myAdvancerequests$.subscribe((res) => {
```
Comment (OmkarJ13) [inline src/app/fyle/my-advances/my-advances.page.spec.ts:226]: lets rename this to myAdvanceRequests$

File: src/app/core/mock-data/extended-advance-request.data.ts [hunk 3]
```diff
@@ -702,3 +702,21 @@ export const allTeamAdvanceRequestsRes: ApiV2Response<ExtendedAdvanceRequest> =
   offset: 0,
   url: '/v2/advance_requests',
 };
+
+export const myAdvancerequestsData2: ApiV2Response<ExtendedAdvanceRequest> = {
```
Comment (Dimple16) [inline src/app/core/mock-data/extended-advance-request.data.ts:706]: `myAdvanceRequestsData2`

**Review by Dimple16 — APPROVED** [review]
minor comment

**Pull request merged**